### DISCUSSION TITLE
warnings when adding DerivativeTwoPhaseMaterial to KKS example

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#16766] [Q&A Modules: Phase field] [answered]
If I add a `DerivativeTwoPhaseMaterial` to a [KKS example problem](https://github.com/idaholab/moose/blob/next/modules/phase_field/examples/kim-kim-suzuki/kks_example_dirichlet.i) using this block:
```
  [./two_phase]
    type = DerivativeTwoPhaseMaterial
    f_name = Fsum
    args = 'cl cs'
    eta = eta
    derivative_order = 2
    fa_name = fl
    fb_name = fs
  [../]    
```
I get some warnings from the kernels when running the problem:
```
*** Warning ***
Missing coupled variables {eta} (add them to args parameter of ChemPotSolute)


*** Warning ***
Missing coupled variables {eta} (add them to args parameter of ChemPotSolute)


*** Warning ***
Missing coupled variables {eta, cs} (add them to args parameter of CHBulk)
```
I'm curious why this is. I see now that the kernels mentioned in the warnings take the single-phase chemical energy materials and do not need `eta`, so the `DerivativeTwoPhaseMaterial` is adding a global dependency on `eta`? Is this something I need to worry about? I wouldn't imagine that adding extra variable dependences would matter (except maybe in terms of performance) but wanted to check that this is expected behavior.

**Comment by laagesen** [comment, marked as answer]
Hey Jesse, these warnings are triggered by the `validateNonlinearCoupling` function in `initialSetup()` for both these kernels. Are you supplying `Fsum` to these kernels after you added it? I'm guessing not but just wanted to make sure.

**Reply by jessecarterMOOSE** [reply]
No I left everything else the same. I'm only making some extra materials for validation/postprocessing reasons.

**Reply by jessecarterMOOSE** [reply]
@laagesen the results come out the same either way, so is it safe to ignore the warnings? Would that always be the case?

**Reply by laagesen** [reply]
Yes it should be safe to ignore it. Although the message is a bit confusing and we will try to look into why it's getting triggered in this situation, which is not desired. In general this warning is only telling you that you are missing coupled variables which might be needed for a calculation of Jacobians. So it should not affect the results you get even in a situation where you are actually missing something. It could potentially make convergence worse, but should not affect the answer you get (assuming convergences tolerances are set to reasonable values).

**Reply by jessecarterMOOSE** [reply]
Makes sense. Thanks!
